markets/perps-market/test/integration/Orders/OffchainAsyncOrder.pending.test.ts

```diff
@@ -0,0 +1,184 @@
+import { snapshotCheckpoint } from '@synthetixio/core-utils/utils/mocha/snapshot';
+import { depositCollateral, settleOrder } from '../helpers';
+import assertBn from '@synthetixio/core-utils/utils/assertions/assert-bignumber';
+import assertRevert from '@synthetixio/core-utils/utils/assertions/assert-revert';
+import { getTxTime } from '@synthetixio/core-utils/src/utils/hardhat/rpc';
+
+describe('Offchain Async Order - Prevent updates with pending order test', () => {
+  const { systems, perpsMarkets, provider, trader1, keeper } = bootstrapMarkets({
+    synthMarkets: [],
+    perpsMarkets: [
+      {
+        name: 'Ether',
+        token: 'snxETH',
+        price: bn(1000),
+        fundingParams: { skewScale: bn(100_000), maxFundingVelocity: bn(10) },
+      },
+      {
+        name: 'Bitcoin',
+        token: 'snxBTC',
+        price: bn(10_000),
+        fundingParams: { skewScale: bn(100_000), maxFundingVelocity: bn(10) },
+      },
+    ],
+    traderAccountIds: [2, 3],
+  });
+  let ethMarketId: ethers.BigNumber;
+  let btcMarketId: ethers.BigNumber;
+
+  before('identify actors', async () => {
+    ethMarketId = perpsMarkets()[0].marketId();
+    btcMarketId = perpsMarkets()[1].marketId();
+  });
+
+  describe('With a pending order', () => {
+    let startTime: number;
+    before('add collateral', async () => {
+      await depositCollateral({
+        systems,
+        trader: trader1,
+        accountId: () => 2,
+        collaterals: [
+          {
+            snxUSDAmount: () => bn(10_000),
+          },
+        ],
+      });
+    });
+
+    before('commit the order', async () => {
+      const tx = await systems()
+        .PerpsMarket.connect(trader1())
+        .commitOrder({
+          marketId: ethMarketId,
+          accountId: 2,
+          sizeDelta: bn(1),
+          settlementStrategyId: 0,
+          acceptablePrice: bn(1050), // 5% slippage
+          trackingCode: ethers.constants.HashZero,
+        });
+      startTime = await getTxTime(provider(), tx);
+    });
+
+    const restoreToCommit = snapshotCheckpoint(provider);
+
+    describe('failures', () => {
+      before(restoreToCommit);
+
+      it('reverts if attempt to update collateral', async () => {
+        await assertRevert(
+          systems().PerpsMarket.connect(trader1()).modifyCollateral(2, 0, bn(10)),
+          'PendingOrdersExist()',
+          systems().PerpsMarket
+        );
+      });
+
+      it('reverts if attempt to commit another order', async () => {
+        await assertRevert(
+          systems()
+            .PerpsMarket.connect(trader1())
+            .commitOrder({
+              marketId: btcMarketId,
+              accountId: 2,
+              sizeDelta: bn(0.01),
+              settlementStrategyId: 0,
+              acceptablePrice: bn(10050), // 5% slippage
+              trackingCode: ethers.constants.HashZero,
+            }),
+          'PendingOrdersExist()',
+          systems().PerpsMarket
+        );
+      });
+    });
```

Comment:
No sure how valuable, but we could add an assertion for an order on the same market too?